Use Dependency Injection

robert-mihaiAdam opened the pull request.

victorgugonatu commented on Domain/Dto/Bets.cs:
```diff
@@ -10,12 +10,17 @@ public class Bets
         [Required]
         public Guid BetableEntityB { get; set; }
+
+        public void SetTime(TimeProvider timeProvider)
+        {
+            Date = timeProvider.GetUtcNow().DateTime;
+        }
```
let's remove this - we can set the property from the service

robert-mihaiAdam replied:
It's private the constructor

victorgugonatu commented on Domain/Dto/PlacedBets.cs:
```diff
@@ -1,4 +1,5 @@
 ﻿using System.ComponentModel.DataAnnotations;
+using static System.Runtime.InteropServices.JavaScript.JSType;
```
??? what is this ?

victorgugonatu commented on Services/BetQuoteServices.cs:
```diff
@@ -7,13 +7,15 @@
 namespace Services
 {
-    public class BetQuoteServices : IBetableService<BetQuotes, UpdateBetQuotes>
+    public sealed class BetQuoteServices : IBetQuoteService
```
```suggestion
    public sealed class BetQuoteService : IBetQuoteService
```
let's keep it in singular

victorgugonatu commented on Services/BetsServices.cs:
```diff
@@ -7,18 +7,22 @@
+            _timeProvider = timeProvider;
         }
 
         private async Task<bool> validateBetBody(Bets entity)
```
```suggestion
        private async Task<bool> ValidateBetBody(Bets entity)
```

also let's append Async to all async methods

victorgugonatu commented on Services/BetsServices.cs:
```diff
@@ -7,18 +7,22 @@
         private async Task<bool> validateBetBody(Bets entity)
         {
-            BetableEntityServices service = new(_dbContext);
+            entity.SetTime(_timeProvider);
```
```suggestion
            entity.Date = _timeProvider.GetUtcNow().DateTime;
```

victorgugonatu commented on BetsApi/Controllers/BetsController.cs:
```diff
@@ -1,35 +1,43 @@
+        private readonly IBetableEntityService betableEntityService;
 
-        public BetsController(DBContext context)
+        public BetsController(DBContext context,
```
DBContext is not needed

victorgugonatu commented on Services/BetQuoteService.cs:
```diff
@@ -0,0 +1,36 @@
+    public sealed class BetQuoteService : IBetQuoteService
+    {
+        private readonly DBContext _dbContext;
+        private readonly TimeProvider _timeProvider;
```
is this used?

victorgugonatu commented on Services/BetableEntityService.cs:
```diff
@@ -0,0 +1,35 @@
+    public sealed class BetableEntityService : IBetableEntityService
+    {
+        private readonly DBContext _dbContext;
+        private readonly TimeProvider _timeProvider;
```
is this used?

victorgugonatu commented on Services/BetsService.cs:
```diff
@@ -0,0 +1,61 @@
+    private async Task<bool> ValidateBetBodyAsync(Bets entity)
+        {
+            entity.Date = _timeProvider.GetUtcNow().DateTime;
+            BetableEntityService service = new(_dbContext, _timeProvider);
```
please inject the service instead of creating the instance manually

victorgugonatu commented on Services/PlacedBetsService.cs:
```diff
@@ -6,51 +6,47 @@
             Guid quoteId = entity.QuoteId;
-            BetQuoteServices betQuoteService = new BetQuoteServices(_dbContext);
-            BetQuotes currentQuote = await betQuoteService.GetById(quoteId);
+            BetQuoteService betQuoteService = new BetQuoteService(_dbContext, _timeProvider);
```
inject the service via ctor

victorgugonatu commented on Seeder/Program.cs:
```diff
@@ -15,55 +15,55 @@ public static void Main(string[] args)
-                context.BetableEntity.Add(entity);
-            }
-            context.SaveChanges();
+            //for (long i = 1; i <= noEntities; i++)
+            //{
+            //    BetableEntityService service = new(context, TimeProvider.System);
+            //    BetableEntity entity = new BetableEntity { Name = $"Echipa {i * 12}" };
+            //    context.BetableEntity.Add(entity);
+            //}
+            //context.SaveChanges();
```
is this code needed or not?
if not, delete it

robert-mihaiAdam committed:
Use Dependency Injection and add TimeProvider in the services:

commit 7fdd275ec108bb8f982a8ede30302c404f9c1abf
Merge: ac74bc5 a148d01
Author: Robert Adam <robert.adam@uipath.com>
Date:   Fri Jul 19 16:41:50 2024 +0300

    Merge branch 'DI' of https://github.com/robert-mihaiAdam/Bets into DI

commit ac74bc54cc64ab7e8c249d3f5a156cead3db0f6d
Author: Robert Adam <robert.adam@uipath.com>
Date:   Fri Jul 19 16:39:08 2024 +0300

    Squashed commit of the following:

    commit e9d31952e9f17d6157800f9a9eb6a6934aece683
    Merge: fe8f0d8 a819170
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Fri Jul 19 16:19:39 2024 +0300

        Solve conflicts caused by rebase

    commit a819170c39bfce735ff539e5bb4b9f5e969d8fd3
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Fri Jul 19 16:11:23 2024 +0300

        Feature: Use Dependency Injection in all services, and add TimeProvider as one

    commit f3387829e29785a0f0f6643a26e4d68c3b86c3a3
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Fri Jul 19 16:09:08 2024 +0300

        Feature: Use Dependency Injection in all services, and add TimeProvider as one

    commit fe8f0d816f701b2127f901bf12090f2e8ee6d103
    Merge: 6dcb4ac 136b9a8
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Fri Jul 19 15:48:28 2024 +0300

        Solve conflicts

    commit 136b9a8bbe73fc3a5e2aed4053ebdc1df69dcc1c
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Fri Jul 19 15:39:40 2024 +0300

        fix: Seeder with the new services

    commit 302397e63a1190f16759efa9910e08b4ae5d703f
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Fri Jul 19 13:20:10 2024 +0300

        feat: Implement Dependency Injection and add TimeProvider instead of setting the Datetime from entity creation

    commit 6dcb4ac2c9439f31451ed05fc335f4d2a75d0dd1
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Fri Jul 19 13:08:02 2024 +0300

        Integrated all services using Dependency Injection and removed unusages services from them

    commit b3b6c7091cdc63ef292233ea1f986b4da684afeb
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Fri Jul 19 10:34:29 2024 +0300

        Remove some unusage imports

    commit 6326ab50477626d986e1acc9422f2c33bd3a2e7b
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Fri Jul 19 10:31:51 2024 +0300

        Group all async methods

    commit 6453e392c32bebaaf1f735f78d5024d721f9732c
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Thu Jul 18 17:27:56 2024 +0300

        Remove genericity

    commit bbe97ffe3efb41399044c0ef63f830e6cf5e40ad
    Merge: 596c5f4 7b5b7fe
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Thu Jul 18 16:46:14 2024 +0300

        Solve conflicts

    commit 596c5f44dc64d036a226eb53a6d2cd1cb9bf6937
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Thu Jul 18 16:22:46 2024 +0300

        Remove generic interface

    commit 7b5b7fe865a1454215d62e8d8c75344c531e6d41
    Merge: 023f8f4 77961ac
    Author: robert-mihaiAdam <robert.adam@uipath.com>
    Date:   Thu Jul 18 16:05:06 2024 +0300

        Merge pull request #5 from robert-mihaiAdam/domain_name

        Change names and ad TimeProvider

    commit 77961ac343fa1f8e5ea8159a93599234770fddcf
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Thu Jul 18 11:43:05 2024 +0300

        Add TimeProvider as DI

    commit 54d4adbe8a5806b9fc707bb1287eb1c283ef914e
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Thu Jul 18 10:12:21 2024 +0300

        Solve sln conflict

    commit 842b2413d141829a39ffc1642b183928d0eaae1e
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Thu Jul 18 09:33:11 2024 +0300

        Add interfaces for every service

    commit ded24a4b71b360c984311c594d6012b9e650778a
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Wed Jul 17 14:04:50 2024 +0300

        Revision DI

    commit e0fdd19aa5ef458025ce35e12a48b09fbfd42fad
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Wed Jul 17 11:37:02 2024 +0300

        Integrate Dependency Injection in Controller

    commit 727af948d6c1cd10b5400f85a9fd888425e2e948
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Wed Jul 17 10:30:47 2024 +0300

        Remove DI

commit a148d0186cb56656d88a999c47a126983fe9781f
Author: Robert Adam <robert.adam@uipath.com>
Date:   Fri Jul 19 16:39:08 2024 +0300

    Squashed commit of the following:

    commit e9d31952e9f17d6157800f9a9eb6a6934aece683
    Merge: fe8f0d8 a819170
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Fri Jul 19 16:19:39 2024 +0300

        Solve conflicts caused by rebase

    commit a819170c39bfce735ff539e5bb4b9f5e969d8fd3
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Fri Jul 19 16:11:23 2024 +0300

        Feature: Use Dependency Injection in all services, and add TimeProvider as one

    commit f3387829e29785a0f0f6643a26e4d68c3b86c3a3
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Fri Jul 19 16:09:08 2024 +0300

        Feature: Use Dependency Injection in all services, and add TimeProvider as one

    commit fe8f0d816f701b2127f901bf12090f2e8ee6d103
    Merge: 6dcb4ac 136b9a8
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Fri Jul 19 15:48:28 2024 +0300

        Solve conflicts

    commit 136b9a8bbe73fc3a5e2aed4053ebdc1df69dcc1c
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Fri Jul 19 15:39:40 2024 +0300

        fix: Seeder with the new services

    commit 302397e63a1190f16759efa9910e08b4ae5d703f
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Fri Jul 19 13:20:10 2024 +0300

        feat: Implement Dependency Injection and add TimeProvider instead of setting the Datetime from entity creation

    commit 6dcb4ac2c9439f31451ed05fc335f4d2a75d0dd1
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Fri Jul 19 13:08:02 2024 +0300

        Integrated all services using Dependency Injection and removed unusages services from them

    commit b3b6c7091cdc63ef292233ea1f986b4da684afeb
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Fri Jul 19 10:34:29 2024 +0300

        Remove some unusage imports

    commit 6326ab50477626d986e1acc9422f2c33bd3a2e7b
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Fri Jul 19 10:31:51 2024 +0300

        Group all async methods

    commit 6453e392c32bebaaf1f735f78d5024d721f9732c
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Thu Jul 18 17:27:56 2024 +0300

        Remove genericity

    commit bbe97ffe3efb41399044c0ef63f830e6cf5e40ad
    Merge: 596c5f4 7b5b7fe
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Thu Jul 18 16:46:14 2024 +0300

        Solve conflicts

    commit 596c5f44dc64d036a226eb53a6d2cd1cb9bf6937
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Thu Jul 18 16:22:46 2024 +0300

        Remove generic interface

    commit 7b5b7fe865a1454215d62e8d8c75344c531e6d41
    Merge: 023f8f4 77961ac
    Author: robert-mihaiAdam <robert.adam@uipath.com>
    Date:   Thu Jul 18 16:05:06 2024 +0300

        Merge pull request #5 from robert-mihaiAdam/domain_name

        Change names and ad TimeProvider

    commit 77961ac343fa1f8e5ea8159a93599234770fddcf
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Thu Jul 18 11:43:05 2024 +0300

        Add TimeProvider as DI

    commit 54d4adbe8a5806b9fc707bb1287eb1c283ef914e
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Thu Jul 18 10:12:21 2024 +0300

        Solve sln conflict

    commit 842b2413d141829a39ffc1642b183928d0eaae1e
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Thu Jul 18 09:33:11 2024 +0300

        Add interfaces for every service

    commit ded24a4b71b360c984311c594d6012b9e650778a
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Wed Jul 17 14:04:50 2024 +0300

        Revision DI

    commit e0fdd19aa5ef458025ce35e12a48b09fbfd42fad
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Wed Jul 17 11:37:02 2024 +0300

        Integrate Dependency Injection in Controller

    commit 727af948d6c1cd10b5400f85a9fd888425e2e948
    Author: Robert Adam <robert.adam@uipath.com>
    Date:   Wed Jul 17 10:30:47 2024 +0300

        Remove DI

robert-mihaiAdam committed:
Fix: remove unusage packages and fix BetableEntityService

victorgugonatu committed:
fixes
